rust/src/delta.rs

```diff
@@ -1454,14 +1502,57 @@ impl<'a> DeltaTransaction<'a> {
         //     IsolationLevel::Serializable
         // };
 
-        let prepared_commit = self.prepare_commit(_operation).await?;
+        let prepared_commit = self.prepare_commit(operation).await?;
 
         // try to commit in a loop in case other writers write the next version first
         let version = self.try_commit_loop(&prepared_commit).await?;
 
         Ok(version)
     }
 
+    /// Commit transaction with additional app metadata
+    pub async fn commit_with_info(
+        &mut self,
+        operation: Option<DeltaOperation>,
+        app_metadata: Option<Map<String, Value>>,
+    ) -> Result<DeltaDataTypeVersion, DeltaTableError> {
```

Comment:
Could you add a TODO here to remind checking operation and decide whether the commit is safe to retry in a loop?